core/src/main/java/com/xatkit/core/recognition/RecognitionMonitor.java

```diff
@@ -1,578 +1,21 @@
-     * @return the created {@link JsonObject}
-     */
-    private JsonObject buildSessionObject(String sessionId, Map<Long, IntentRecord> sessionData) {
-        JsonObject sessionObject = new JsonObject();
-        sessionObject.addProperty("sessionId", sessionId);
-        JsonArray sessionRecords = new JsonArray();
-        sessionObject.add("entries", sessionRecords);
-        int unmatchedCount = 0;
-        int matchedCount = 0;
-        double accConfidence = 0.0;
-        for (Map.Entry<Long, IntentRecord> sessionEntry : sessionData.entrySet()) {
-            JsonObject entryObject = new JsonObject();
-            sessionRecords.add(entryObject);
-            entryObject.addProperty("timestamp", sessionEntry.getKey());
-            entryObject.addProperty("utterance", sessionEntry.getValue().getUtterance());
-            entryObject.addProperty("intent", sessionEntry.getValue().getIntentName());
-            entryObject.addProperty("confidence", sessionEntry.getValue().getRecognitionConfidence());
-            if (sessionEntry.getValue().getIntentName().equals("Default_Fallback_Intent")) {
-                unmatchedCount++;
-            } else {
-                accConfidence += sessionEntry.getValue().getRecognitionConfidence();
-                matchedCount++;
-            }
-        }
-        sessionObject.add("matchedUtteranceCount", new JsonPrimitive(matchedCount));
-        sessionObject.add("unmatchedUtteranceCount", new JsonPrimitive(unmatchedCount));
-        if (matchedCount > 0) {
-            sessionObject.add("avgSessionConfidence",
-                    new JsonPrimitive(accConfidence / (double) matchedCount));
-        }
-        return sessionObject;
-    }
 
+public interface RecognitionMonitor {
+    
     /**
      * Logs the recognition information from the provided {@code recognizedIntent} and {@code session}.
      *
      * @param session          the {@link XatkitSession} from which the {@link RecognizedIntent} has been created
      * @param recognizedIntent the {@link RecognizedIntent} to log
      */
-    public void logRecognizedIntent(XatkitSession session, RecognizedIntent recognizedIntent) {
-        Long ts = System.currentTimeMillis();
-        Map<Long, IntentRecord> sessionMap = records.get(session.getSessionId());
-        if (isNull(sessionMap)) {
-            sessionMap = new TreeMap<>();
-        }
-        sessionMap.put(ts, new IntentRecord(recognizedIntent));
-        records.put(session.getSessionId(), sessionMap);
-        db.commit();
-    }
-
+    void logRecognizedIntent(XatkitSession session, RecognizedIntent intent);
 
     /**
-     * Commit the pending operations on the database and closes the connection.
+     * If MAPDB: Commit the pending operations on the database and closes the connection. 
+     * If INFLUXDB: Closes connection to database. Changes should be commited, but check influxDB doc in case some actions need to be performed!
```

Comment:
Do not reference subclasses (or specific implementations) at the interface level, this should be done in the corresponding class' documentation.
I'll take care of it.